### PR TITLE
Issue 41 - Refactor GitHubServiceFactory as an EJB/ CDI bean

### LINKED ISSUE #41: Refactor GitHubServiceFactory as an EJB/ CDI bean
Let's get rid of a lot of the custom ServiceLoader code and rely upon the container to inject an instance.

Tests should be done in stages: a base that has an extension to provide the GitHubServiceFactory as a POJO, and another test to run against the container getting the GitHubServiceFactory injected as an EJB or CDI managed component

### PR DESCRIPTION
**Pull request by xcoulon**
Removed the GitHubServiceLoader class and the META-INF/service file
Using a CDI producer to instanciate the GitHubService implementation
Added integration test with Arquillian (deploying the GitHubService
within a war file, along with all required libs). The integration test
class extends the POJO test class to avoid code duplication.

The IT test may be run with the following command line:

    mvn clean verify -Pit,wildfly-managed-arquillian \
      -DGITHUB_TOKEN=$GITHUB_TOKEN_CATAPULT -DGITHUB_USERNAME=$GITHUB_USERNAME

where `GITHUB_TOKEN_CATAPULT` and `GITHUB_USERNAME` are the user's token
for the Catapult application, and his/her GitHub username.

Fixes #41

### CODE REVIEW

File: services/github-service-impl/pom.xml [hunk 1]
```diff
@@ -1,51 +1,108 @@
...
+                    <execution>
+                        <goals>
+                            <goal>test-jar</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+    <profiles>
```
Comment (ALRubinger) [inline services/github-service-impl/pom.xml]: Check the "tests" module; we skip all of this by using arquillian-chameleon! :)

Comment (xcoulon) [inline services/github-service-impl/pom.xml]: ok, let me check that. I'm an old-school Arquillian user and never used Chameleon before ;-)
As far as code organization is concerned, should I move the integration tests for `GitHubService` into the `tests` modules, then ?

Comment (xcoulon) [inline services/github-service-impl/pom.xml]: also, do we want to run tests in `remote` mode only, or in `managed` mode as well ?

File: services/github-service-impl/src/main/java/org/kontinuity/catapult/service/github/utils/SystemUtils.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
```
Comment (ALRubinger) [inline services/github-service-impl/src/main/java/org/kontinuity/catapult/service/github/utils/SystemUtils.java]: Not EPL. :)  Gotta get this template out of yo' Workspace for Catapult :)

File: services/github-service-impl/src/test/resources/arquillian.xml [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
-   <container qualifier="arquillian-wildfly-managed" />
-   <container qualifier="arquillian-wildfly-remote" />
+<arquillian xmlns="http://jboss.org/schema/arquillian"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://jboss.org/schema/arquillian http://jboss.org/schema/arquillian/arquillian_1_0.xsd">
+	<container qualifier="arquillian-wildfly-managed" default="true">
+		<configuration>
+			<property name="jbossHome">${JBOSS_HOME}</property>
+			<!-- pass the GitHub token and username system properties from Maven CLI (or Eclipse) to WildFly -->
+			<property name="javaVmArguments">-DGITHUB_TOKEN=${GITHUB_TOKEN} -DGITHUB_USERNAME=${GITHUB_USERNAME}</property>
```
Comment (ALRubinger) [inline services/github-service-impl/src/test/resources/arquillian.xml]: Why pass as system props if they're already available as env vars?

File: services/github-service-impl/pom.xml (hunk 1 above)
Comment (ALRubinger) [inline services/github-service-impl/pom.xml]: Code organization: the "tests" module is for tests against the full EAR (or any other artifacts) as built by Maven.  If you want to run tests with microdeployments (as you define them using ShrinkWrap), then you can do them here in this module.  It's about the level of integration you wanna hit. :)

Comment (ALRubinger) [inline services/github-service-impl/pom.xml]: Yeah we should support remote and managed both (check how we configured it in "tests" module using Chameleon.  It's so awesome I'll never go back to defining my container explicitly again).

**Comment by ALRubinger** [comment]
Just a note: I know I made line comments, but I'll likely pull this in this weekend and massage it up to get it upstream.  You don't need to work over the weekend of course.  I just have some catchup to do as I lost the last 2 days to meetings. :)

**Comment by ALRubinger** [comment]
Upstream: 
b984ec7c25fac1f51e0ffe2aedd4641f79372fe3
c4e2a711971232272de4c916ca2950d21a6f6f77

@xcoulon, you may wanna check my commit just to see how Chameleon is applied, and the `GitHubCredentials` class is already there to abstract the envvar/sysprop stuff.  Also I put the test logic in a base class, and made extensions to run it as both an integration test and a unit test (just a little different from having the IT extend from the Unit).

Great stuff, glad we got this in there.

**Comment by xcoulon** [comment]
@ALRubinger thanks for your feedback on the PR and thanks for your comments. Looks like I have to learn more about Arquillian Chameleon ;-)

**Pull request closed without merging**